x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/mergers/rule_source/calculate_rule_source_from_asset.test.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RuleResponse } from '../../../../../../../../common/api/detection_engine';
+import type { PrebuiltRuleAsset } from '../../../../../prebuilt_rules';
+import { calculateRuleSourceFromAsset } from './calculate_rule_source_from_asset';
+
+const buildTestRule = (overrides?: Partial<RuleResponse>) => {
+  return {
+    rule_id: 'rule_id',
+    version: 1,
+    ...overrides,
+  } as RuleResponse;
+};
+
+const buildTestRuleAsset = (overrides?: Partial<PrebuiltRuleAsset>) => {
+  return {
+    rule_id: 'rule_id',
+    version: 1,
+    source_updated_at: '2024-05-01',
+    ...overrides,
+  } as PrebuiltRuleAsset;
+};
+
+describe('calculateRuleSourceFromAsset', () => {
+  it('calculates as internal if no asset is found', () => {
+    const result = calculateRuleSourceFromAsset({
+      rule: buildTestRule(),
+      prebuiltRuleAsset: undefined,
+      ruleIdExists: false,
+    });
+
+    expect(result).toEqual({
+      type: 'internal',
+    });
+  });
+
+  it('calculates as unmodified external type if an asset is found without a matching version', () => {
+    const result = calculateRuleSourceFromAsset({
+      rule: buildTestRule(),
+      prebuiltRuleAsset: buildTestRuleAsset({ version: 2 }),
+      ruleIdExists: true,
+    });
+
+    expect(result).toEqual({
+      type: 'external',
+      is_customized: false,
+    });
+  });
```

Comment:
⚠️ **IMPORTANT!**

This test actually made me realize that we should most probably rethink how we handle this scenario where an asset is found, but there's no matching version. Right now, the RFC says that we should set `is_customized` to `false`. I'm do not remember exactly what was the reason for that decision semantically speaking.

But anyways: playing around with these tests, I couldn't get this one to pass. The `calculateRuleSourceFromAsset` method returned `is_customized: true` even though all fields in the `rule` and `prebuiltRuleAsset` were equal. But then I realized: all are equal **except for the version**.

And the problem with that is that `calculateRuleSourceFromAsset` uses `calculateIsCustomized` under the hood, which returns `true` if **any** fields differ in the two rules passed to it, including the version. This works for our current use case because we always compare a Rule and a PrebuiltRuleAsset of the same version, but in this use case we can compare different versions. And therefore we are always getting `is_customized: true`.

So I think we should correct what was written in the RFC, and this function should look so:
```ts
export const calculateRuleSourceFromAsset = ({
  rule,
  prebuiltRuleAsset,
  ruleIdExists,
}: {
  rule: RuleResponse | PrebuiltRuleToImport;
  prebuiltRuleAsset: PrebuiltRuleAsset | undefined;
  ruleIdExists: boolean;
}): RuleSource => {
  // No PrebuiltRuleAsset found with same rule_id
  if (!ruleIdExists) {
    return {
      type: 'internal',
    };
  }

  // PrebuiltRuleAsset found with same rule_id, but different version
  if (prebuiltRuleAsset == null) {
    return {
      type: 'external',
      is_customized: true, // changed here from false to true, differs from RFC
    };
  }

  // PrebuiltRuleAsset with matching rule_id and version found
  const isCustomized = calculateIsCustomized(prebuiltRuleAsset, rule);

  return {
    type: 'external',
    is_customized: isCustomized,
  };
};
```

Reply:
I agree, and I've made this change in b47153225492d5fcaa5b703d14fb3e7248a03ee6.